server/filestore.go

```diff
@@ -2370,7 +2370,38 @@ func (fs *fileStore) Purge() uint64 {
 		cb(-int64(purged), -rbytes, 0, _EMPTY_)
 	}
 
-	return purged
+	return purged, nil
+}
+
+// Compact will remove all messages from this store up to
+// but not including the seq parameter.
+// Will return the number of purged messages.
+func (fs *fileStore) Compact(seq uint64) (uint64, error) {
+	if seq == 0 {
+		return fs.Purge()
+	}
+
+	if _, err := fs.msgForSeq(seq); err != nil {
+		return 0, err
+	}
+
+	var purged uint64
+	for fseq := fs.firstSeq(); fseq < seq; fseq = fs.firstSeq() {
```

Comment:
So is the caller responsible to make sure that `seq` is valid? What would happen if I call Compact(100) on a file that has 1 to 10. Will we get to try to removeMsg(11) and that will fail with not found (or similar error)? If so, it means that Compact() did remove messages but still returns a failure. It maybe ok since the caller asked to remove up to 100 (not included) and that functions only removed 1 to 10. But just making sure that this is what you expect.

Reply:
We check that it is valid on line 2384 so would get an error right away. However what you asked me to add, checking the error the way I do is wrong. I need to skip across these messages if they are simply not there. The code before did that since it only cared if we removed the message. 

Let me add more error code handling to piece out different errors.

Reply:
ok I think this is correct now.

Reply:
Once we are good will squash.